internal/batches/service/service.go

```diff
@@ -442,13 +442,17 @@ func (svc *Service) ResolveNamespace(ctx context.Context, namespace string) (str
 }
 
 func (svc *Service) ResolveRepositories(ctx context.Context, spec *batcheslib.BatchSpec) ([]*graphql.Repository, error) {
-	seen := map[string]*graphql.Repository{}
+	type seenRepo struct {
+		overridable bool
+		revisions   []*graphql.Repository
+	}
+	seen := map[string]*seenRepo{}
```

Comment:
This is obviously eerily similar to code in https://github.com/sourcegraph/sourcegraph/pull/28168 — my not-so-secret plan here is to unify these once we have generics available in Go, since the only real distinction is that the "revision" type is different in `src` and SSBC.